config.js

```diff
@@ -0,0 +1,56 @@
+"use strict";
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+const fs = require("fs"),
+      path = require("path"),
+      yaml = require('js-yaml');
+
+
+//------------------------------------------------------------------------------
+// Private
+//------------------------------------------------------------------------------
+
+const CONFIG_FILE = ".speccy.yml";
+const filePath = path.join(__dirname + '/' + CONFIG_FILE);
+
+//------------------------------------------------------------------------------
+// Public Interface
+//------------------------------------------------------------------------------
+
+// Determines if config file exists
+function fileExists() {
+  if (fs.existsSync(filePath)) {
+    return true;
+  } else {
+    return false;
+  }
+}
+
+// Returns config object if config file found
+function getConfigFile() {
+  if (fileExists()) {
+    var config = yaml.safeLoad(fs.readFileSync(filePath, 'utf8'), { json: true });
+    return config;
+  }
+}
+
+// Returns array of rules file(s) from config file
+function getConfigRules() {
+  const config = getConfigFile();
+  let rulesFiles = [];
+
+  config.lint.rules.forEach(function(rule) {
```

Comment:
This line will fail if the config file doesn't have `config.lint`, or `config.lint.rules`.

Also, the forEach then push is just copying the array, so you could just do:

```
const rulesFiles = config.lint.rules;
return rulesFiles;
```

or

```
return config.lint.rules;
```

Reply:
Maybe we could make things a bit more generic, as there are a lot of config items, and we don't want methods for each of them. 

https://www.npmjs.com/package/config

If that config system could be configured to load `.speccy.yml` then maybe we could just use it. 

Then the line that grabs the rules would just look like this:

```
const rules = [].concat(config.get('lint.rules'), cmd.rules);
await loader.loadRuleFiles(cmd.rules, { verbose });
```

Reply:
The config npm package is definitely the better way to go here. At present, the config package reads from a `default.EXT` file. I've been unsuccessful at configuring it to read from `.speccy.yml` as it's eluding me. However, was able to set the `NODE_CONFIG_DIR` to `./.speccy/` temporarily until the custom file name is sorted.